cranelift-filetests/src/test_unwind.rs

```diff
@@ -0,0 +1,233 @@
+    assert_eq!(parsed.command, "unwind");
+    if !parsed.options.is_empty() {
+        Err(format!("No options allowed on {}", parsed))
+    } else {
+        Ok(Box::new(TestUnwind))
+    }
+}
+
+impl SubTest for TestUnwind {
+    fn name(&self) -> &'static str {
+        "unwind"
+    }
+
+    fn is_mutating(&self) -> bool {
+        false
+    }
+
+    fn needs_isa(&self) -> bool {
+        true
+    }
+
+    fn run(&self, func: Cow<ir::Function>, context: &Context) -> SubtestResult<()> {
+        let isa = context.isa.expect("unwind needs an ISA");
+        let mut comp_ctx = cranelift_codegen::Context::for_function(func.into_owned());
+
+        comp_ctx.compile(isa).expect("failed to compile function");
+
+        let mut mem = Vec::new();
+        comp_ctx.emit_unwind_info(isa, &mut mem);
+
+        let mut text = String::new();
+        if mem.len() == 0 {
+            writeln!(text, "No unwind information.").unwrap();
+        } else {
+            print_unwind_info(&mut text, &mem);
+        }
+
+        run_filecheck(&text, context)
+    }
+}
+
+fn print_unwind_info(text: &mut String, mem: &[u8]) {
+    assert!(mem.len() >= core::mem::size_of::<UnwindInfo>());
+    let info = unsafe { UnwindInfo::from_ptr(mem.as_ptr()) };
+
+    let node_count = info.unwind_code_count();
+
+    // Assert correct alignment and padding of the unwind information
+    assert!(mem.len() % 4 == 0);
+    assert_eq!(
+        mem.len(),
+        core::mem::size_of::<UnwindInfo>()
+            + ((node_count as usize) * core::mem::size_of::<UnwindCode>())
+            + if (node_count & 1) == 1 {
+                core::mem::size_of::<UnwindCode>()
+            } else {
+                0
+            }
+    );
+
+    writeln!(
+        text,
+        "Unwind info: version = {}, flags = {}, prologue size: {}, unwind code count = {}, frame register = {}, frame register offset = {}",
+        info.version(),
+        info.flags(),
+        info.prologue_size(),
+        node_count,
+        info.frame_register(),
+        info.frame_register_offset()
+    ).unwrap();
+
+    let mut i = 0;
+    while i < node_count {
+        let offset = (core::mem::size_of::<UnwindInfo>()
+            + ((i as usize) * core::mem::size_of::<UnwindCode>())) as isize;
+        let code = unsafe { UnwindCode::from_ptr(mem.as_ptr().offset(offset)) };
+
+        write!(
+            text,
+            "Unwind code: offset = {}, operation = {:#?}, info = {}",
+            code.offset(),
+            code.operation(),
+            code.info(),
+        )
+        .unwrap();
+
+        match code.additional_nodes() {
+            0 => {
+                writeln!(text).unwrap();
+                i += 1;
+            }
+            1 => {
+                let offset = offset + core::mem::size_of::<UnwindCode>() as isize;
+                writeln!(text, ", value = {}", unsafe {
+                    core::ptr::read_unaligned(mem.as_ptr().offset(offset) as *const u16)
```

Comment:
Please use `LittleEndian::read_u16(&mem[offset..offset+2])` instead. (LittleEndian comes from the byteorder crate)

Reply:
I didn't see a dependency on `byteorder` yet, so I was reluctant to add it, but that's what I will likely use.